File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/POLIIMORFISMO/EJERCICIO 1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="Sin nombre", genero="Desconocido", jugadores=0):
        self.nombre = nombre
        self.genero = genero
        self.jugadores = jugadores

    def mostrar(self):
        print(f"Nombre: {self.nombre}, Género: {self.genero}, Jugadores: {self.jugadores}")

    # Sobrecarga mediante argumentos opcionales
    def agregar_jugadores(self, cantidad=1):
        self.jugadores += cantidad


def main():
    v1 = Videojuego("Minecraft")
    v2 = Videojuego("FIFA", "Deportes", 4)

    v1.agregar_jugadores()        # agrega 1 jugador
    v2.agregar_jugadores(3)       # agrega 3 jugadores

    v1.mostrar()
    v2.mostrar()

if __name__ == "__main__":
    main()

File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/POLIIMORFISMO/EJERCICIO 1/Videojuego/src/Ejercicio_1/Videojuego.java
package Ejercicio_1;

public class Videojuego {
	
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;
    
    public Videojuego() {
        this.nombre = "Desconocido";
        this.plataforma = "No especificada";
        this.cantidadJugadores = 0;
    }
    
    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 0;
    }
    
    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }
    
    public void mostrar() {
        System.out.println("Nombre del juego: " + nombre);
        System.out.println("Plataforma: " + plataforma);
        System.out.println("Cantidad de jugadores: " + cantidadJugadores);
    }
    
    public void agregarJugadores() {
        this.cantidadJugadores += 1;
    }
    
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
    }
}
